bf2fc6cc711aee1a0c2a/kas-fleet-manager
# feat(prewarming): make it possible for manual clusters to make use of pre-warming

## Description


Create a dedicated pre-warming configuration that's generic enough and can be used irrespective of whether dynamic scaling is turned on or not. 
This allows for manual clusters to make use of the pre-warming feature as well. 

The configuration looks like
```yaml
---
developer:
  reserved_streaming_units: m # if 0 then no pre-warming
  base_streaming_unit_size: x1 
standard:
  reserved_streaming_units: n # if 0 then no pre-warming
  base_streaming_unit_size: x1
```
where `m` and `n` can be integers. 
The `base_streaming_unit_size` allows to configure the size and it can be `x1`, `x2` etc depending on which size is available.
During startup, a validation is perfomed to make sure that the size and instance types are available, otherwise we throw an error.
We explicitly allow for the no configurability of pre-warming as it is an optional feature i.e if the file is not configured or does not exist, we do not throw an error during startup.

**NOTE**: The default behavior is that no pre-warming will be applied. The default value in service template is an empty object i.e {} and in development config, the instance types are there with `0` for reserved streaming units field.

Closes https://issues.redhat.com/browse/MGDSTRM-9235

## Verification Steps


1. Modified tests should pass
2. Against a real cluster verify that pre-warming managed kafkas are created irrespective of the scaling mode.
For this, make sure that the pre-warming configuration have been setup for each instance type.

## Checklist (Definition of Done)
